### PR TITLE
adds /ban, /unban, /baninfo, /banlist, /role, /mod, and /admin commands

### PR DESCRIPTION
**Pull request by substack**
depends on https://github.com/cabal-club/cabal-core/pull/80

I've tested some scenarios on a local cabal (but not all). I was able to add an admin, see their ban (with /banlist) and see the ban disappear with /unban.

### CODE REVIEW

**Comment by cblgh** [comment]
@substack do you have anything against swapping out `/ban`, `/unban` etc for `/hide` `/unhide` (or `/show`)? as per 
* https://gist.github.com/cblgh/617e01f851a3b2cb0417200b88ba4960
* https://gist.github.com/cblgh/ef643b1f35511ae86dcf262b8717cc5f

it shouldn't change anything while still keeping the semantics intact as we add `/block`, `/mute` etc

File: src/commands.js [hunk 1]
```diff
@@ -173,7 +175,180 @@ module.exports = {
...
+        next()
+      }
+    }
+  },
+  admin: {
+    help: () => 'add or remove an admin for a channel or cabal-wide',
+    call: (cabal, res, arg) => {
+      if (!arg) {
+        res.info('usage: /admin (add|remove) (CHANNEL|@) KEY {REASON...}')
+        res.info('usage: /admin list (CHANNEL|@)')
```
Comment (cblgh) [inline src/commands.js:314]: can we test out adding a mod/admin cabal-wide as the only option, to begin with? 

it feels like there's a lot going on already, so having granular mods might be too much to keep track of. (i'm fine with that granularity in cabal-core)

it seems to me that the default is normally going to be cabal-wide

**Comment by substack** [comment]
These can be changed to whatever. The bans aren't even used yet in the moderation views, I was just going off of how the other PR added cabal.ban(). We could also add the other methods that the draft command guide uses, this was mostly to flesh out how those would all be implemented.

**Comment by cblgh** [comment]
yeah that's what i thought, just figured one easy win would be to use `/hide` since that's the effect of a ban atm

also the prs are ofc AMAZING <3

**Comment by substack** [comment]
closing this as #39 takes in these changes and adds more

**Pull request closed without merging**